src/spac/visualization.py

```diff
@@ -128,34 +169,19 @@ def visualize_2D_scatter(
                         fontsize=9, ha='center', va='center'
                     )
 
-            # Extract current handles and labels for the legend
-            handles, labels = ax.get_legend_handles_labels()
-
-            # Adjust the size of legend items
-            for handle in handles:
-                handle.set_sizes([legend_label_size])
-
-            # Create a custom legend
-            ax.legend(
-                handles,
-                labels,
-                loc='upper right',
-                bbox_to_anchor=(1.25, 1),  # Adjusting position
-                handlelength=2,
-                handletextpad=1

```

Comment:
this is a source of failure for another unittest.

specifically, test_legend_placement_categorical_labels is failing. if you've changed how you handle legends (specifically where you put it), you need to update or remove that unittest.